scipy/integrate/tests/test_quadrature.py

```diff
@@ -325,6 +317,33 @@ def test_ndim(self, xp):
         r = trapezoid(q, x=z, axis=2)
         xp_assert_close(r, qz)
 
+    @skip_xp_backends('jax.numpy',
+                      reasons=["JAX arrays do not support item assignment"])
+    @pytest.mark.usefixtures("skip_xp_backends")
+    def test_gh21908(self, xp):
+        # extended testing for n-dim arrays
+        x = xp.reshape(xp.linspace(0, 29, 30), (3, 10))
+        y = xp.reshape(xp.linspace(0, 29, 30), (3, 10))
+
+        out0 = xp.linspace(200, 380, 10)
+        xp_assert_close(trapezoid(y, x=x, axis=0), out0)
+        assert_allclose(trapezoid(y, x=np.array([0, 10., 20.]), axis=0), out0)
+        # x needs to be broadcastable against y
+        assert_allclose(
+            trapezoid(y, x=np.array([0, 10., 20.])[:, None], axis=0),
+            out0
+        )
+        with pytest.raises(Exception):
+            # x is not broadcastable against y
+            trapezoid(y, x=np.array([0, 10., 20.])[None, :], axis=0)
+
+        out1 = np.array([ 40.5, 130.5, 220.5])
+        assert_allclose(trapezoid(y, x=x, axis=1), out1)
+        assert_allclose(
+            trapezoid(y, x=np.array(np.linspace(0, 9, 10)), axis=1),
+            out1
+        )
```

Comment:
in case I wasn't clear, the rest of the test needs changes too (`np` -> `xp`, `assert_allclose` -> `xp_assert_close`)